ddr/lib/ddr-scanner/dwarf/DwarfParser.cpp

```diff
@@ -79,9 +79,51 @@ dwarf_init(int fd,
 	 */
 	FILE *fp = NULL;
 	if (DW_DLV_OK == ret) {
-		stringstream ss;
-		ss << "dwarfdump " << filepath << " 2>&1";
-		fp = popen(ss.str().c_str(), "r");
+
+		/* On some Mac machines, calling "dwarfdump" in shell calls a version named
+		 * "llvm-dwarfdump" instead of the version of dwarfdump DDR uses, which is
+		 * instead named "dwarfdump-classic" on those systems. Thus, we first try to
+		 * find "dwarfdump-classic" directly in case "dwarfdump" is linked to
+		 * "llvm-dwarfdump". If we don't find "dwarfdump-classic", then the machine
+		 * we are on uses the correct version of dwarfdump.
+		 */
+		stringstream toolpath;
+		fp = popen("xcrun -f dwarfdump-classic 2>&1", "r");
+		if (NULL != fp) {
+			char *buffer = NULL;
+			size_t cap = 0;
+			size_t len = getline(&buffer, &cap, fp);
+			if ((-1 == len) || (buffer[0] != '/')) {
```

Comment:
`getline()` returns `ssize_t`, not `size_t`.

Reply:
The coding standard prefers `'/' != buffer[0]` over `buffer[0] != '/'`.
Rather than make assumptions, I think this should check `NULL != buffer` before dereferencing that pointer.
These comments apply here and in similar situations below.